tests/nested/manual/muinstaller-core/task.yaml

```diff
@@ -0,0 +1,212 @@
+  cp -a ./"$SEED_DIR"/system-seed/ /var/lib/snapd/seed
+
+  # build the muinstaller snap
+  snap install snapcraft --candidate --classic
+  "$TESTSTOOLS"/lxd-state prepare-snap
+  (cd "$TESTSLIB"/muinstaller && snapcraft)
+  MUINSTALLER_SNAP="$(find "$TESTSLIB"/muinstaller/ -maxdepth 1 -name '*.snap')"
+  echo "found $MUINSTALLER_SNAP"
+
+  # create new disk for the installer to work on and attach to VM
+  truncate --size=4G fake-disk.img
+
+  # create a VM and mount a cloud image
+  tests.nested build-image classic
+
+  # TODO: nested classic images do not support secure boot today so
+  #       this will not work to test the secure boot installer. So for
+  #       now the workaround is to boot classic to create user/ssh
+  #       keys, shutdown down, convert disk from qcow2->raw and rename
+  #       from classic->core and use nested_start_core_vm (like below)
+  #
+  # start it so that cloud-init creates ssh keys and user
+  # We set a serial for our disk to easily locate it when invoking muinstaller (virtio-target)
+  NESTED_PARAM_EXTRA="-drive file=$(pwd)/fake-disk.img,if=none,snapshot=off,format=raw,id=disk2 \
+      -device virtio-blk-pci,drive=disk2,serial=target"
+  tests.nested create-vm classic --extra-param "$NESTED_PARAM_EXTRA"
+
+  # make sure classic image is bootable with snakeoil keys
+  # TODO: move to nested_create_classic_image
+  # XXX: use assets from gadget instead?
+  for s in BOOT/BOOTX64.EFI ubuntu/shimx64.efi; do
+      remote.exec "sudo cp -a /boot/efi/EFI/$s /tmp"
+      remote.exec "sudo chmod 755 /tmp/$(basename $s)"
+      remote.pull /tmp/"$(basename $s)" .
+      nested_secboot_sign_file "$(basename $s)" "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+      remote.push "$(basename $s)"
+      remote.exec "sudo mv $(basename $s) /boot/efi/EFI/$s"
+  done
+
+  remote.exec "sudo sh -c 'echo SNAPD_DEBUG=1 >> /etc/environment'"
+  # push our snap down
+  remote.push "$SPREAD_PATH"/../snapd_*.deb
+  remote.exec "sudo apt install -y ./snapd_*.deb"
+
+  # push our seed down
+  # TODO: merge with classic /var/lib/snapd/seed eventually
+  # XXX: port scp -r to remote.push
+  #remote.push ./"$SEED_DIR"/system-seed/ '~/'
+  sshpass -p ubuntu scp -r -P 8022 -o ConnectTimeout=10 \
+      -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no \
+      ./"$SEED_DIR"/system-seed/ user1@localhost:~/install-seed
+  remote.exec "sudo mv /home/user1/install-seed /var/lib/snapd/"
+
+  # shutdown the classic vm to install with a core VM that supports
+  # secboot/tpm
+  tests.nested vm stop
+  sync
+
+  # HACK: convert "classic" qcow2 to raw "core" image because we need
+  # to boot with OVMF we really should fix this so that classic and
+  # core VMs are more similar
+  qemu-img convert -f qcow2 -O raw \
+      "$NESTED_IMAGES_DIR/$(nested_get_image_name classic)" \
+      "$NESTED_IMAGES_DIR/$(nested_get_image_name core)"
+  # and we don't need the classic image anymore
+  # TODO: uncomment
```

Comment:
Should this be cleaned up before the merge or is it already equally messy in musinstaller-classic?

Reply:
Done